crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2177,12 +2196,14 @@ impl World {
         B: Bundle<Effect: NoBundleEffect>,
     {
         self.flush();
-
         let change_tick = self.change_tick();
 
+        // SAFETY: These come from the same world.
+        let mut registrator =
+            unsafe { ComponentsRegistrator::new(&mut self.components, &mut self.component_ids) };
```

Comment:
Not using `.component_registrator()` was confusing here. Can we add a comment to all of these cases (there are several) to explain "We can't use `.component_registrator()` since we want to mutably borrow other fields in the world".